bpf/lib/nodeport.h

```diff
@@ -188,13 +188,19 @@ static __always_inline int nodeport_nat_ipv6_fwd(struct __ctx_buff *ctx,
 		.max_port = NODEPORT_PORT_MAX_NAT,
 	};
 	int ret;
+	__u32 monitor = 0;
 
 	ipv6_addr_copy(&target.addr, addr);
 
 	ret = snat_v6_needed(ctx, addr) ?
 	      snat_v6_process(ctx, NAT_DIR_EGRESS, &target) : CTX_ACT_OK;
+
 	if (ret == NAT_PUNT_TO_STACK)
-		ret = CTX_ACT_OK;
+		return CTX_ACT_OK;
+
+	monitor = ctx_load_meta(ctx, CB_CT_MONITOR);
+	send_trace_notify(ctx, TRACE_TO_NETWORK, 0, 0, 0, 0, ret, monitor);
```

Comment:
`send_trace_notify` is often provided a `src_id` argument. How can this be obtained from a function like `nodeport_nat_ipv6_fwd`? I've seen `src_id = ipcache_lookup_srcid{4,6}(...)`, but those lookups look like they're quite lengthy (instruction-wise). Would they be worth including here?

Reply:
I *think* we can follow the same idea of placing it into the skb metadata as what I had suggested with the monitor value, functions like to_netdev() should already have resolved the source ID so it's just a matter of transferring that into the skb during the potential tail call, reading it back out in `tail_handle_nat_fwd_ip*()` and passing it along.